src/fontra/backends/designspace.py

```diff
@@ -459,14 +474,22 @@ async def putGlyph(
             glyphSet = ufoLayer.glyphSet
             writeGlyphSetContents = glyphName not in glyphSet
 
+            customData = None
             if glyphSet == self.defaultUFOLayer.glyphSet:
                 layerGlyph = defaultLayerGlyph
                 storeInLib(layerGlyph, GLYPH_DESIGNSPACE_LIB_KEY, localDS)
                 storeInLib(layerGlyph, SOURCE_NAME_MAPPING_LIB_KEY, sourceNameMapping)
                 storeInLib(layerGlyph, LAYER_NAME_MAPPING_LIB_KEY, layerNameMapping)
-                storeInLib(layerGlyph, CUSTOM_DATA_LIB_KEY, glyph.customData)
+                customData = (
+                    glyph.customData | localSourcesCustomData[ufoLayer.fontraLayerName]
+                )
             else:
                 layerGlyph = readGlyphOrCreate(glyphSet, glyphName, codePoints)
+                if localSourcesCustomData.get(ufoLayer.fontraLayerName) is not None:
```

Comment:
General Python advice:
- To check if a key exists in a dict, you write "if key in dict"
- To get a key with a fallback, use "dict.get(key)"

You _either_ first check if a key exists, and then retrieve it normally, or you "get" it and then check the value for None. You are using a mix of both here.

Reply:
Actually, you just moved it in [this commit](https://github.com/googlefonts/fontra/pull/1403/commits/34e934aa51a03837d783ee84d5ae00100cb9d086)... I fixed it (together with a general logic error) [here](https://github.com/googlefonts/fontra/pull/1403/commits/7ff723a36ed16c4a22dd417022e89dc713801ef5).